theano/configdefaults.py

```diff
@@ -1505,6 +1505,15 @@ def filter_vm_lazy(val):
              EnumStr('regular', 'fast'),
              in_c_key=False)
 
+AddConfigVar('check_stack_trace',
+             "A flag for checking the stack trace during the optimization"
+             " process. The default is not checking the stack trace of any"
+             " optimization. check_all checks if the stack trace is added"
+             " and check_and_skip checks all the stack trace of optimizations"
+             " but skips the ones that have a dummy value ",
+             EnumStr('not_checking', 'check_and_skip', 'check_all'),
```

Comment:
Maybe we should rename the options a little bit. I would suggest:
- `"off"`: do not perform any check (default)
- `"log"` (or maybe `"insert_dummy"? not sure about how to name it): inserts a dummy stack trace that identifies the optimization that inserted that variable
- `"warn"`: prints a warning if a stack trace is missing
- `"raise"`: raises an exception if a stack trace is missing